ts/screens/authentication/IdpLoginScreen.tsx

```diff
@@ -96,32 +104,84 @@ const onNavigationStateChange = (
 class IdpLoginScreen extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props);
-
     this.state = {
-      requestState: pot.noneLoading
+      requestState: pot.noneLoading,
+      startingMs: new Date().getTime()
     };
   }
 
-  private handleOnError = (): void =>
+  private getSpidId = (): string =>
+    this.props.loggedOutWithIdpAuth
+      ? this.props.loggedOutWithIdpAuth.idp.entityID
+      : "idpID n/a";
+
+  public componentDidMount() {
+    this.props.dispatch(
```

Comment:
don't call `this.props.dispatch` directly, define methods in `mapDispatchToProps` instead